### ISSUE TITLE
Searching state is not clear when switch to people cards page from person detail page

### ISSUE THREAD

**Issue by Fog3211** [stakwork/sphinx-tribes-frontend#152]
**Describe the bug**
When performing search filtering on the user details page, if the search criteria is not cleared and the "back" button is clicked to return to the previous `/p` page, the searching conditions are not reset.

**To Reproduce**
1. Go to [people page](https://community.sphinx.chat/p)
2. Search user by some keywords
3. Click `Back` button without clear the search keywords
4. See the wrong search result

https://github.com/stakwork/sphinx-tribes-frontend/assets/23151576/1dd1a603-a4ae-415d-a635-26eee377a15f

**Expected behavior**
Searching conditions should be reset when switch to people cards page

**Desktop (please complete the following information):**
 - Browser Chrome 120

### Acceptance Criteria
- [x] I have tested on Chrome desktop
- [x] I have posted a screenshot or video in my PR

**Comment by ecurrencyhodler** [comment]
This should be fixed on staging. If it's not, ping me and we can reopen. Thanks for QA-ing the site!

**Comment by Fog3211** [comment]
@ecurrencyhodler 
It appears that the issue still persists in the staging environment. To rule out the possibility of the problem being caused by incomplete code deployment to the staging environment, I replicated the issue locally using the latest code from the master branch, and the result was the same.

https://github.com/stakwork/sphinx-tribes-frontend/assets/23151576/95d99e38-2bf2-45bc-85be-117fc30b3297

**Comment by Fog3211** [comment]
I found the link to the issue you mentioned earlier, and it appears that in the previous bug fix, only the value in the search bar was cleared, but it did not trigger an update of the search results.

https://github.com/stakwork/sphinx-tribes-frontend/pull/18#issuecomment-1913958171

Additionally, I'm not the only one who has noticed this issue. I saw that `MahtabBukhari` also raised this issue #99  after the pull request was made.
